app/(candidate)/dashboard/outreach/components/OutreachCreateCards.js

```diff
@@ -139,6 +147,21 @@ export default function OutreachCreateCards({ tcrCompliance }) {
         }}
       />
 
+      <P2PUpgradeModal
+        {...{
+          variant: (() => {
+            if (!isPro) return P2P_MODAL_VARIANTS.NonProUpgrade
+            if (p2pUxEnabled && hasFreeTextsOffer && !isTextCompliant) {
+              return P2P_MODAL_VARIANTS.ProFreeTextsNonCompliant
+            }
+            return P2P_MODAL_VARIANTS.NonProUpgrade
+          })(),
+          open: showP2PModal,
+          onClose: () => setShowP2PModal(false),
+          onUpgradeLinkClick: undefined,
+        }}
```

Comment:
This variant logic is being duplicated across multiple usages of `P2PUpgradeModal`